homeassistant/components/binary_sensor/vera.py

```diff
@@ -0,0 +1,95 @@
+"""
+Support for Vera binary sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.vera/
+"""
+import logging
+
+import homeassistant.util.dt as dt_util
+from homeassistant.const import (
+    ATTR_ARMED, ATTR_BATTERY_LEVEL, ATTR_LAST_TRIP_TIME, ATTR_TRIPPED)
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice)
+from homeassistant.loader import get_component
+
+REQUIREMENTS = ['pyvera==0.2.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Perform the setup for Vera controller devices."""
+    vera = get_component('vera')
+    devices = vera.VERA_DEVICES[vera.BINARY_SENSOR]
+    controller = vera.VERA_CONTROLLER
+
+    sensors = [VeraBinarySensor(device, controller)
+               for device in devices]
+    add_devices_callback(sensors)
+
+
+class VeraBinarySensor(BinarySensorDevice):
+    """Representation of a Vera Binary Sensor."""
+
+    def __init__(self, vera_device, controller):
```

Comment:
This init looks very generic. Maybe you want to move it to a `VeraDevice` class and use partial inheritance? See the Z-Wave devices for an example